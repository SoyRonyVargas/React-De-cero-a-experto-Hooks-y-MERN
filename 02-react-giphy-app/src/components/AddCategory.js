import React, { useState } from 'react'
import PropTypes from 'prop-types';

export const AddCategory = ({ setCategories }) => {

    const [inputValue, setInputValue] = useState(''); // ''

    const handleInputChange = ( e ) => {
        
        console.log('handle llamado');

        setInputValue( e.target.value );

    }

    const handleSubmit = (e) => {
        
        e.preventDefault();

        console.log('formulario prevenido' , inputValue);

        if ( inputValue.trim().length > 2 ) {
            setCategories( cats => [ inputValue, ...cats, ] );
            setInputValue('');
        }

    }

    return (
        <form onSubmit={ handleSubmit }>
            <p id="valor"> { inputValue } </p>
            <input 
                type="text"
                value={ inputValue }
                onChange={ handleInputChange }
            />
        </form>
    )
}


AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}
