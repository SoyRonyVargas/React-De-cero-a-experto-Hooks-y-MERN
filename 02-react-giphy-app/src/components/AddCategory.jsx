import React, { useState } from 'react'
import PropTypes from 'prop-types';

export const AddCategory = ({ setCategories }) => {

    const [inputValue, setInputValue] = useState('');

    const handleInputChange = ( e ) => {
        
        setInputValue( e.target.value );

    }

    const handleSubmit = (e) => {
        
        e.preventDefault();

        if ( inputValue.trim().length > 2 ) {
            setCategories( cats => [ inputValue, ...cats, ] );
            setInputValue('');
        }

    }

    return (
        <form aria-label='form' onSubmit={ handleSubmit }>
            <p id="valor"> { inputValue } </p>
            <input 
                type="text"
                value={ inputValue }
                onChange={ handleInputChange }
            />
        </form>
    )
}


AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}
