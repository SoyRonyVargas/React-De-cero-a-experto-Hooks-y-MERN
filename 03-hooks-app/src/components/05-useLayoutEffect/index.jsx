import useCounter from "../../hooks/useCounter";
import useFetch from "../../hooks/useFetch";
import React , { useLayoutEffect , useRef } from "react";

const MultipleHooks = () => {
  const { counter, increment } = useCounter(1);

  const { data, loading } = useFetch(
    `https://www.breakingbadapi.com/api/quotes/${counter}`
  );

  const { quote } = !!data && data[0];

  const ref = useRef();

  useLayoutEffect( () => {

    console.log(ref.current.getBoundingClientRect());

  }, [quote])

  return (
    <div className="">
      <h1>useLayoutEffect</h1>
      <hr />
      <figure className="card p-3">
        <blockquote className="blockquote">
          <p ref={ref}> {quote} </p>
        </blockquote>
      </figure>
      <button onClick={increment} className="btn btn-primary">
        Next Quote
      </button>
    </div>
  );
};

export default MultipleHooks;
