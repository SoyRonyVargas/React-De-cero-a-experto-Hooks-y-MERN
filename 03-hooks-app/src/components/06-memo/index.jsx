import { useState } from "react";
import useCounter from "../../hooks/useCounter"
import Small from './Small'

const MultipleHooks = () => {
  
    const { counter, increment } = useCounter(1);
    const [ show, setShow ] = useState(false);

    return (
        <div className="">
        <h1>Memo</h1>
        <Small value={counter} />
        <hr />
        
        <button onClick={increment} className="btn btn-primary mx-1">
            Next 
        </button>
        <button onClick={() => {
            setShow(!show)
        }} className="btn btn-primary">
            Show / Hide { JSON.stringify(show , null , 3)} 
        </button>
        </div>
    );
};

export default MultipleHooks;
