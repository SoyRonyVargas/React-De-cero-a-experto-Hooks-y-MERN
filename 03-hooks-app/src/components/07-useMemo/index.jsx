import { useMemo, useState } from "react";
import useCounter from "../../hooks/useCounter"

const MultipleHooks = () => {
  
    const { counter, increment } = useCounter(1);
    const [ show, setShow ] = useState(false);

    const process = ( it = 1 ) => {

        for( let i = 0 ; i < it ; i++ )
        {
            console.log('iteracion...');
        }

        return it + ' iteraciones realizadas.'

    }

    const memoProcess = useMemo( () => process(counter) , [ counter ] )	

    return (
        <div className="">
            
            <h1>useMemo {counter} </h1>
            
            <hr />

            <p>
                { memoProcess }
            </p>
            
            <button onClick={increment} className="btn btn-primary mx-1">
                Next 
            </button>
            <button onClick={() => {
                setShow(!show)
            }} className="btn btn-primary">
                Show / Hide { JSON.stringify(show , null , 3)} 
            </button>
        </div>
    );
};

export default MultipleHooks;
