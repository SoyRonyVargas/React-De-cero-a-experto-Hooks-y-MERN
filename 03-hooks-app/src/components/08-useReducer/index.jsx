import React, { useReducer , useRef } from 'react'
import TodoList from './TodoList'
import TodoAdd from './TodoAdd'
import reducer from './reducer'

const index = () => {

    const ref = useRef()

    const initialState = [
        {
            id: new Date().getTime(),
            desc: 'Aprender React',
            done: false
        }
    ]

    const [ todos , dispatch ] = useReducer(reducer, initialState)

    const handleSubmit = (e) => {

        e.preventDefault()

        const newTodo = {
            id: new Date().getTime(),
            desc: ref.current.value,
            done: false
        }

        ref.current.value = ''

        const action = {
            type: 'ADD_TODO',
            payload: newTodo
        }

        dispatch(action)

    }

    const handleDelete = ( id ) => {

        const action = {
            type: 'delete',
            payload: id
        }

        dispatch(action)

    }

    return (
        <div>
            <h1> Todos - {todos.length} </h1>
            <hr />
            <main className='row '>
                
                <TodoList
                    handleDelete={handleDelete}
                    todos={todos}
                />

                <TodoAdd
                    handleSubmit={handleSubmit}
                    ref={ref}
                />

            </main>
        </div>
    )

}

export default index