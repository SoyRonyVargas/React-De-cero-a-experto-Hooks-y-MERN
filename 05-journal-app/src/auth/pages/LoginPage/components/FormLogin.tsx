import { Button, Grid, TextField, Typography, Link , Alert } from "@mui/material"
import { selectStatus , selectError } from "../../../../store/auth/authSlice"
import { Link as RouterLink } from 'react-router-dom'
import { useForm } from "../../../../hooks/useForm" 
import { useAppDispatch } from "../../../../hooks"
import { Google } from "@mui/icons-material"
import { useSelector } from 'react-redux'
import { 
    checkAuthWithEmailAndPassword, 
    checkGoogleAuth 
} from "../../../../store/auth/thunks"

const formValidators = {
    email: [ ( value : string ) => value.includes("@") && value.trim().length > 0 , "Ingresa un correo electronico valido"],
    password: [ ( value : string ) => value.length > 0 , "Ingresa la contraseña"],
}

const initialState = {
    email: "",
    password: "",
}

const FormLogin = () => {

    const dispatch = useAppDispatch()

    const isChecking = useSelector(selectStatus)
    const errorLogin = useSelector(selectError)

    const handleGoogleSignIn = () => {

        dispatch(checkGoogleAuth())        

    }

    const { 
        onInputChange,
        formState,
        errors ,
        submited,
        handleSubmitForm,
        isFormValid
    } = useForm( initialState , formValidators )

    const handleSubmit = () => {

        handleSubmitForm()

        if( !isFormValid ) return;

        dispatch(checkAuthWithEmailAndPassword({
            ...formState,
            displayName: ""
        }))

    }

    return (
        <form className="w-100 animate__animated animate__fadeIn animate__faster">
            
            <img 
                src="/images/logo.svg" 
                className="logo__standard" 
                alt="Logo" 
            />

            <Grid container>
                <Grid item
                    lg={12}
                    xs={12}
                    md={12}
                    sx={{
                        mt: 2
                    }}
                >
                    <TextField
                        placeholder="correo@gmail.com"
                        disabled={isChecking}
                        variant="outlined"
                        name="email"
                        label="Correo"
                        fullWidth
                        error={!!errors?.email && submited}
                        onChange={onInputChange}
                        value={formState.email}
                        helperText={(!!errors?.email && submited) && errors?.email }
                    />
                </Grid>
                <Grid item
                    xs={12}
                    sx={{
                        mt: 2
                    }}
                >
                    <TextField
                        type="password"
                        placeholder="Tu contraseña"
                        disabled={isChecking}
                        label="Contraseña"
                        variant="outlined"
                        fullWidth
                        name="password"
                        error={!!errors?.password && submited}
                        onChange={onInputChange}
                        value={formState.password}
                        helperText={(!!errors?.password && submited) && errors?.password }
                    />
                </Grid>

            </Grid>

            {
                errorLogin &&
                <Alert sx={{ mt: 2 }} severity="error">
                    { errorLogin }
                </Alert>
            }

            <Grid
                container
                spacing={2}
                sx={{ mb: 2 }}
            >

                <Grid item xs={12} sm={6} sx={{ mt: 2 }}>
                    <Button 
                        onClick={handleSubmit}
                        disabled={isChecking}
                        variant="contained" 
                        color="secondary" 
                        fullWidth
                    >
                        <Typography sx={{ ml: .5, textTransform: 'capitalize' }}>
                            Iniciar Sesión
                        </Typography>
                    </Button>
                </Grid>

                <Grid item xs={12} sm={6} sx={{ mt: 2 }}>
                    <Button 
                        onClick={handleGoogleSignIn} 
                        disabled={isChecking}
                        variant="contained" 
                        color="primary" 
                        fullWidth
                    >
                        <Google color={"secondary"} />
                        <Typography sx={{ ml: .5, textTransform: 'capitalize' }}>
                            Google
                        </Typography>
                    </Button>
                </Grid>

            </Grid>

            <Grid container sx={{ mt: 3 }}>
                <Grid item xs>
                    <Link component={RouterLink} to="/">

                    </Link>
                </Grid>
                <Grid item>
                    <Link component={RouterLink} to="/auth/register">
                        {"¿No tienes cuenta? Registrate"}
                    </Link>
                </Grid>
            </Grid>

        </form>
    )
}

export default FormLogin