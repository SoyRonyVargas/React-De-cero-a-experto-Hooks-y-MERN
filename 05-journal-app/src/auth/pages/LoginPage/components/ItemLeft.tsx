import { Grid } from "@mui/material"

const ItemLeft = () => {
  return (
    <Grid
        item
        xs={false}
        sm={4}
        md={7}
        container
        spacing={0}
        direction="column"
        alignItems="center"
        justifyContent="center"
        sx={{
          backgroundColor: "#1A1928"
        }}
      >
        <img src="/images/login3.png" className="logo__standard" alt="" />
      </Grid>
  )
}

export default ItemLeft