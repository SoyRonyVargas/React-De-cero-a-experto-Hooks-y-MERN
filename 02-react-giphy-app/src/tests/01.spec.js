import { GifGridItem } from '../components/GifGridItem'
import { render , screen } from '@testing-library/react'
import React from 'react'

describe('Test a la app gif', () => {
    
    const title = 'Titulo'
    
    const url = 'https://media1.giphy.com/media/l0HlJz7y7zqQkQq9m/giphy.gif'
    
    test('Tomando snapshot del componente', () => {
        
        const { container } = render(<GifGridItem title={title} url={url} />)      

        expect(container).toMatchSnapshot()

    })
    
    test('debe contener la imagen el url', () => {
    
        render(<GifGridItem title={title} url={url} />)      

        // expect(screen.getByRole("img").src).toBe(url)
        
        // expect(screen.getByRole("img").alt).toBe(title)
        
        const { src , alt } = screen.getByRole("img")

        expect(src).toBe(url)
        
        expect(alt).toBe(title)
        
    })
    
    test('Debe de mostrar el titulo en el componente', () => { 

        render(<GifGridItem title={title} url={url} />)      

        expect( screen.getByText(title) ).toBeTruthy()

    })

    // test('Debe obtener la imagen y su url y alt', () => {
        
    //     const img = wrapper.find('img')

    //     const props_img = img.props()

    //     expect(props_img.src).toBe(url)

    //     expect(props_img.alt).toBe(title)

    // })
    
    // test('Deberia tener la clase fade in', () => {
        
    //     const props = wrapper.find('div').props()

    //     const classes = props.className.split(' ')

    //     const has_class = classes.some( className => className == 'animate__fadeIn') 

    //     expect(has_class).toBe(true)

    // })
    

})
