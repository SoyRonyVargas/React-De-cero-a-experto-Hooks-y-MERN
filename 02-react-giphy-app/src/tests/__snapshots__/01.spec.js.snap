// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test a la app gif Tomando snapshot del componente 1`] = `
<div>
  <div
    class="card animate__animated animate__fadeIn"
  >
    <img
      alt="Titulo"
      src="https://media1.giphy.com/media/l0HlJz7y7zqQkQq9m/giphy.gif"
    />
    <p>
       
      Titulo
       
    </p>
  </div>
</div>
`;
